tsl/src/compression/compression.h

```diff
@@ -62,6 +62,109 @@ typedef struct DecompressResult
 	bool is_done;
 } DecompressResult;
 
+/*
+ * Use the Arrow C data interface which is a well-known standard for in-memory
+ * interchange of columnar data.
+ *
+ * https://arrow.apache.org/docs/format/CDataInterface.html
+ */
+typedef struct ArrowArray
```

Comment:
Maybe we could move the `ArrowArray` to its own header file? At the moment we are using it only for compression. But it's a more generic data structure that could be re-used in the future by other implementations.

Reply:
Done.